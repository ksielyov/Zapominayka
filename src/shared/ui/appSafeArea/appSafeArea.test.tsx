import React, {FunctionComponent} from 'react';
import renderer from 'react-test-renderer';
import {AppSafeArea} from './index';
import {View} from 'react-native';
import {SafeAreaProvider} from 'react-native-safe-area-context';
import {render} from '@testing-library/react-native';

const VirtualProvider: FunctionComponent = ({children}) => (
  <SafeAreaProvider>{children}</SafeAreaProvider>
);

test('appSafeArea should render correctly', () => {
  const tree = renderer
    .create(
      <VirtualProvider>
        <AppSafeArea>
          <View />
        </AppSafeArea>
      </VirtualProvider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('appSafeArea should have view nested', () => {
  const testID = 'nestedView';

  const {queryByTestId} = render(
    <VirtualProvider>
      <AppSafeArea>
        <View testID={testID} />
      </AppSafeArea>
    </VirtualProvider>,
  );
  const element = queryByTestId(testID);
  expect(element).toBeTruthy();
});
