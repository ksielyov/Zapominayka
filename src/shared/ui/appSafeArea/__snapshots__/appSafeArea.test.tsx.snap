// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`appSafeArea should render correctly 1`] = `
<View
  style={
    Object {
      "height": "100%",
      "paddingTop": 0,
      "width": "100%",
    }
  }
>
  <View />
</View>
`;
